### DISCUSSION TITLE
如何设置fcflags位置

### DISCUSSION THREAD

**Discussion by xiangyi-wang** [xmake-io/xmake#2332] [Q&A] [answered]
fcflags`lnetcdff`需要在编译文件后面，如何设置fcflags的位置顺序呢
正确顺序
`gfortran -o xy -I/home/wxy/Build_WRF/LIBRARIES/openmpi/netcdf-4.7.4/include/  ./src/simple_xy_wr.f90 -lnetcdff -L/home/wxy/Build_WRF/LIBRARIES/openmpi/netcdf-4.7.4/lib/ `
错误顺序
`gfortran -o xy -I/home/wxy/Build_WRF/LIBRARIES/openmpi/netcdf-4.7.4/include/  -lnetcdff ./src/simple_xy_wr.f90 -L/home/wxy/Build_WRF/LIBRARIES/openmpi/netcdf-4.7.4/lib/ `
![image](https://user-images.githubusercontent.com/22881957/166442667-89ce5394-7921-4658-a769-99ccb7e896cf.png)

**Comment by waruqi** [comment, marked as answer]
flags 跟源文件没啥关系，这个调不了，目前格式 flags sourcefile 这是固定的

**Reply by waruqi** [reply]
另外，-l 是 linker flags ，你加到 compiler flags 里干啥

**Reply by xiangyi-wang** [reply]
。。明白了 谢谢
